queue/adaptive_order.go

```diff
@@ -263,15 +266,31 @@ func (q *adaptiveLocalOrdering) Next(ctx context.Context) amboy.Job {
 
 				if len(items.ready) > 0 {
 					id, items.ready = items.ready[0], items.ready[1:]
-					ret <- items.jobs[id]
+					j := items.jobs[id]
+					if err := q.scopes.Acquire(id, j.Scopes()); err != nil {
+						items.ready = append(items.ready)
+						misses++
+						timer.Reset(time.Duration(misses * rand.Int63n(int64(time.Millisecond))))
```

Comment:
The timer should be reset to 0 if err is nil.

Reply:
I think not actually... the property of missing a lock, and then having a little bit of jitter (not much really) is probably good. the max sleep is 10ms